### ISSUE TITLE
Incorrect Group by syntax passes validation and executes 

### ISSUE THREAD

**Issue by osawyerr** [dolthub/go-mysql-server#1057]
Hi,

it appears that GMS doesn't correctly validate incorrect GROUP BY queries. For example the SQL below is executed by GMS but its incorrect SQL syntax (i.e. its missing the group by clause).

select C_MKTSEGMENT, count(*) from customer;

explain plain gives - 
+--------------------------------------------------------+
| GroupBy                                                |
|  ├─ SelectedExprs(customer.C_MKTSEGMENT, COUNT(*))     
|  ├─ Grouping()                                         |
|  └─ Projected table access on [C_MKTSEGMENT]           
|      └─ Table(customer)                                |
+--------------------------------------------------------+

FYI the underlying the table implements both sql.ProjectedTable and sql.FilteredTable. That probably doesn't matter because the syntax checker should catch the incorrect syntax before it hits the table.

Cheers.

**Comment by timsehn** [comment]
I'm confused. 

Yhis is valid SQL?

`select C_MKTSEGMENT, count(*) from customer;`

It seems the `count(*)` causes our analyzer to produce and empty `GROUPING`. For instance:
```
mysql> explain select count(*) from test;
+------------------------------+
| plan                         |
+------------------------------+
| GroupBy                      |
|  ├─ SelectedExprs(COUNT(*))  |
|  ├─ Grouping()               |
|  └─ Exchange(parallelism=20) |
|      └─ Table(test)          |
+------------------------------+
```

I don't think this is a bug. What am I missing?

**Comment by osawyerr** [comment]
The group by clause is missing. When you do an aggregate like that, you need a group by clause. Type a similar statement in any sql editor/database and you'll see that it gives an error.
________________________________

**Comment by max-hoffman** [comment]
We do no enforce `sql_mode=only_full_group_by` at the moment. We infer a default in the above query, the same as if you set `SET sql_mode = '';` in mysql.

A feature request could be implementing `sql_mode=only_full_group_by`.

**Comment by osawyerr** [comment]
Hmm that’s bizarre. I didn’t know that MySQL let you turn that off either. Most other databases see it as incorrect SQL syntax. Also, since the result ends up being incorrect, it’s quite bizarre that it’s a “feature”.

**Comment by max-hoffman** [comment]
Yeah I'm not sure I'd characterize the existing state as a "feature," just a natural way of initially building GROUP_BY. What happens above is the same thing that happens for WINDOW -- non-aggregates are bundled into the grouped execution block with a default `LAST` expression. If there is no ORDER_BY specified, the result for the column projection is nondeterministic/arbitrary/what have you. With ordering and narrower groupings the result will be deterministic. This is true for MySQL and Dolt in the absence of strict mode.

Windows and group bys currently have less attention than overall execution speed, but if you have a use case where you would like `sql_mode=only_full_group_by` we can add it to the roadmap.

**Comment by osawyerr** [comment]
Ah makes sense. Thanks for your explanation. Yes please add it to your roadmap. It’s pretty low priority I’d say since properly formatted group by queries behaves correctly. It was just abit jarring when I stumbled upon it, so thought I’d raise it in case you weren’t aware of it.

Cheers.
